compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -481,7 +477,14 @@ impl<'a, 'tcx> Instantiator<'a, 'tcx> {
                             }) => {
                                 (may_define_opaque_type(tcx, parent_def_id, opaque_hir_id), origin)
                             }
-                            _ => (def_scope_default(), hir::OpaqueTyOrigin::TyAlias),
```

Comment:
This may be a leftover from impl-trait-in-bindings that we overlooked. We only have impl trait in return position and type alias impl trait, there are no other places that generate `ty::Opaque`, thus I concluded that this arm is unreachable.